modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc

```diff
@@ -0,0 +1,128 @@
+= Migrating User Key-based Encryption to Master Key-based Encryption
+
+== Why Should I Move Away From User Key-based Encryption?
+
+NOTE: User key-based encryption is planned to be removed from ownCloud in the near future. While it is a bit more secure than a central encryption approach, User key-based encryption has some disadvantages. It blocks some additional functions such as the integration of an online editor like LibreOffice or OnlyOffice into ownCloud and can cause problems when sharing files with groups. Therefore Master-key-based encryption is now the recommended setup for all new insallations.
+
+The decryption workflow described here works for both MySQL and SQLite based databases. It will only work when users have enabled the password recovery. Moreover, an admin recovery password needs to be activated and available for the ownCloud administrator.
+
+If you need to migrate from User Key-based to Master Key-based encryption, there are several steps that you need to follow to ensure a smooth and complete transition:
+
+. xref:disable-user-key-based-encryption[Disable User Key-based encryption]
+. xref:remove-the-encryption-records-from-the-owncloud-database[Remove the encryption records from the ownCloud database]
+. xref:remove-the-files_encryption-directory[Remove the `files_encryption` directory]
+. xref:encrypt-the-filesystem-using-master-key-based-encryption[Encrypt the filesystem using Master Key-based encryption]
+
+
+[[disable-user-key-based-encryption]]
+== Disable User Key-based Encryption
+
+The first part of the migration process is to decrypt all files and to disable encryption in ownCloud, which requires three commands to be executed.
+These commands are: xref:configuration/server/occ_command.adoc#encryption[`occ encryption:decrypt-all` and `occ encryption:disable`], and xref::configuration/server/occ_command.adoc#apps-commands[`occ app:disable`].
+
+You can see an example of calling the commands listed below, configured to require no user interaction.
+
+[source,console]
+----
+sudo -u www-data php occ encryption:decrypt-all --continue=yes && \
```

Comment:
The docs uses [AsciiDoc attributes](https://asciidoctor.org/docs/user-manual/#attributes) to help ensure consistency across the documentation. For that reason, please update this example as follows:

```asciidoc
[source,console,subs="attributes+"]
 ----
 {occ-command-example-prefix} encryption:decrypt-all --continue=yes && \
   {occ-command-example-prefix} encryption:disable --no-interaction && \
   {occ-command-example-prefix} app:disable --no-interaction encryption
 ----
```

At build time, in the example above, `{occ-command-example-prefix}` will be replaced by `sudo -u www-data php occ`. This helps keep all examples consistent. For example, should the decision be made in the future to remove `sudo` or how the occ command is called, then the attribute can be updated and all examples will reflect that change. This attribute is defined in site.yml in the `asciidoc -> attributes` section.

Reply:
cool ;) ...